### PR TITLE
Remove: work directory in github action.

### PR DESCRIPTION
**Pull request by xuezhaojun**

### CODE REVIEW

File: .github/workflows/go-presubmit.yml [hunk 1]
```diff
@@ -23,25 +19,19 @@ jobs:
...
+
   verify-deps:
     name: verify-deps
     runs-on: ubuntu-latest
     steps:
       - name: checkout code
         uses: actions/checkout@v3
-        with:
-          fetch-depth: 1
-          path: go/src/open-cluster-management.io/registration
```
Comment (skeeey) [inline .github/workflows/go-presubmit.yml:44]: we may keep this to avoid clone the whole repo?

File: .github/workflows/go-presubmit.yml [hunk 2]
```diff
@@ -23,25 +19,19 @@ jobs:
     steps:
       - name: checkout code
         uses: actions/checkout@v3
-        with:
-          fetch-depth: 1
```
Comment (xuezhaojun) [inline .github/workflows/go-presubmit.yml:27]: The default `fetch-depth` is 0. https://github.com/actions/checkout/blob/ac593985615ec2ede58e132d2e21d2b1cbd6127c/README.md?plain=1#L78

File: .github/workflows/go-presubmit.yml (hunk 1 above)
Comment (xuezhaojun) [reply .github/workflows/go-presubmit.yml:44]: The default fetch-depth is also 1. https://github.com/actions/checkout/blob/ac593985615ec2ede58e132d2e21d2b1cbd6127c/README.md?plain=1#L78

**Pull request merged**